### PR TITLE
ports/mimxrt: Add support for GPIO control of SNVS pins. 

### PR DESCRIPTION
**Pull request by iabdalkader**

### CODE REVIEW

File: ports/mimxrt/boards/MIMXRT1062_af.csv [hunk 1]
```diff
@@ -123,3 +123,6 @@ GPIO_SD_B1_08,USDHC2_DATA4,FLEXSPI_A_DATA0,LPUART7_TXD,SAI1_TX_BCLK,LPSPI2_SOUT,
 GPIO_SD_B1_09,USDHC2_DATA5,FLEXSPI_A_DATA1,LPUART7_RXD,SAI1_TX_SYNC,LPSPI2_SIN,GPIO3_IO09,,,,,,,ALT5
 GPIO_SD_B1_10,USDHC2_DATA6,FLEXSPI_A_DATA2,LPUART2_RXD,LPI2C2_SDA,LPSPI2_PCS2,GPIO3_IO10,,,,,,,ALT5
 GPIO_SD_B1_11,USDHC2_DATA7,FLEXSPI_A_DATA3,LPUART2_TXD,LPI2C2_SCL,LPSPI2_PCS3,GPIO3_IO11,,,,,,,ALT5
+WAKEUP,,,,,,GPIO5_IO00,,NMI,,,,,ALT5
+PMIC_ON_REQ,SNVS_PMIC_ON_REQ,,,,,GPIO5_IO01,,,,,,,ALT0
+PMIC_STBY_REQ,CCM_PMIC_STBY_REQ,,,,,GPIO5_IO02,,,,,,,ALT0
```
Comment (dpgeorge) [inline ports/mimxrt/boards/MIMXRT1062_af.csv:128]: Are these real GPIO pins, should they start with `GPIO_`?

I see it's called `WAKEUP_DIG` in one of the other boards.

File: ports/mimxrt/boards/make-pins.py [hunk 2]
```diff
@@ -118,10 +121,17 @@ def print(self):
...
+                "WAKEUP": "PIN_SNVS",
+                "PMIC_ON_REQ": "PIN_SNVS",
+                "PMIC_STBY_REQ": "PIN_SNVS",
+            }
+
             print(
                 "const machine_pin_obj_t pin_{0} = {1}({0}, {2}, {3}, pin_{0}_af, {4}, {5});\n".format(
                     self.name,
-                    "PIN_LPSR" if "LPSR" in self.name else "PIN",
+                    next((options[value] for value in options if value in self.name), "PIN"),
```
Comment (dpgeorge) [inline ports/mimxrt/boards/make-pins.py:134]: This line is confusing... maybe `options.get(self.name, "PIN")`, and then make sure all possible LPSR names are included in the `options` dict.

File: ports/mimxrt/boards/MIMXRT1062_af.csv (hunk 1 above)
Comment (iabdalkader) [reply ports/mimxrt/boards/MIMXRT1062_af.csv:128]: > Are these real GPIO pins, should they start with `GPIO_`?

Yes those are connected to physical pins, and they support GPIO function, but no they don't start with `GPIO`, the names used for all pins in this file is the pad name from the datasheet (the other ones just happen to start with `GPIO`.

> I see it's called `WAKEUP_DIG` in one of the other boards.

`WAKEUP` is the correct pad name used in the RT1060/2 reference manuals.

File: ports/mimxrt/boards/make-pins.py (hunk 2 above)
Comment (iabdalkader) [reply ports/mimxrt/boards/make-pins.py:134]: > and then make sure all possible LPSR names are included in the `options` dict.

All possible LPSR pads are at least 10, are you sure you want to add them all ?

Comment (dpgeorge) [reply ports/mimxrt/boards/make-pins.py:134]: I think it's cleaner to add all LPSR pads.  That way it's explicit what's supported and you'll get an error if you have a typo in a pin name.

Comment (robert-hh) [reply ports/mimxrt/boards/make-pins.py:134]: The MIMXRT1176_af.csv has already the LPSR entries, and the other chips do not seem to have LPSR pads as such. At least I did not find the in the RM.

Comment (iabdalkader) [reply ports/mimxrt/boards/make-pins.py:134]: Yes I'll will make the change suggested by @dpgeorge but I noticed it fails to parse the 1176 AF file, will need a small fix.

Comment (robert-hh) [reply ports/mimxrt/boards/make-pins.py:134]: What fails?

Comment (iabdalkader) [reply ports/mimxrt/boards/make-pins.py:134]: @robert-hh 
```
line 269, in parse_af_file
    iomux_pin_config[pin.name][af_idx]["inputRegister"].strip("U"),
    ~~~~~~~~~~~~~~~~^^^^^^^^^^
KeyError: 'GPIO_SNVS_00'
```

We need a regex to parse `IOMUXC_GPIO_SNVS_xx` for 117x now that SNVS GPIOs are not skipped, but I think the pad names in the CSV should match the ones in the FSL header.

File: ports/mimxrt/boards/MIMXRT1062_af.csv (hunk 1 above)
Comment (iabdalkader) [reply ports/mimxrt/boards/MIMXRT1062_af.csv:128]: For 117x there's `_DIG` suffix, which has been taken care of.

**Pull request merged**